lib/src/r_tree/r_tree.dart

```diff
@@ -334,5 +339,7 @@ class RTree<E> {
     NonLeafNode<E> newRoot = NonLeafNode<E>(_branchFactor, initialChildNodes: [node1, node2]);
     newRoot.height = _root.height + 1;
     _root = newRoot;
+    node1.parent = _root;
+    node2.parent = _root;
```

Comment:
`_growTree` is invoked when the root is split, either through an individual insertion or bulk-load/insertion. In either case we should be updating each subtree's `parent` to point to the new root.